### ISSUE TITLE
Support for Usage In Monorepos

### ISSUE THREAD

**Issue by garygcchiu** [morrowdigital/watermelondb-expo-plugin#36]
I have my project structure in a typical monorepo structure like the following:

- `repo/`
  - `apps/`
    - `expo/`
      - `app/`
      - `assets/`
      - `.gitignore`
      - `app.config.ts`
      - `app-env.d.ts`
      - `babel.config.js`
      - `eas.json`
      - `index.js`
      - `metro.config.js`
      - `package.json`
    - `...`
  - `packages/`
    - `app/`
    - `db/`
    - `lib/`
  - `package.json`
  - `tsconfig.json`
  - `...`

I believe the paths to dependencies are hardcoded to be in `../node_modules/`: https://github.com/morrowdigital/watermelondb-expo-plugin/blob/main/src/withWatermelon.ts#L250-L252
```ts
  pod 'WatermelonDB', :path => '../node_modules/@nozbe/watermelondb'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi', :modular_headers => true
  pod 'simdjson', path: '../node_modules/@nozbe/simdjson', :modular_headers => true\n\n  `;
```

This would lead to the following issue during the build, since in a Yarn monorepo workspace, dependencies are hoisted higher together in the hierarchy and this error would occur locally and via EAS Build (with or without dev client enabled):

```
➜ npx expo prebuild --platform ios                                                                              
✔ Created native directory
✔ Updated package.json | no changes
» ios: icon: This is the image that your app uses on your home screen, you will need to configure it manually.
✔ Finished prebuild
⚠️  Something went wrong running `pod install` in the `ios` directory.
Command `pod install` failed.
└─ Cause: There are multiple dependencies with different sources for `simdjson` in `Podfile`:

- simdjson (from `../../../node_modules/@nozbe/simdjson`)
- simdjson (from `../node_modules/@nozbe/simdjson`)

pod install --repo-update --ansi exited with non-zero code: 1
```

Failed build from EAS: 
![image](https://github.com/morrowdigital/watermelondb-expo-plugin/assets/19521762/eb9af08d-89cb-465d-af10-e2da25e7d11a)

I think some people would like to develop in a monorepo structure, an example here using WatermelonDB + Expo + monorepo: [[ShareMyStack](https://github.com/bndkt/sharemystack/)] 

Is there a way to specify the path for the workspace root, (i.e. `../../` instead of `./`)? Or a way to pass the paths to the dependencies another way? 

Thank you!

**Comment by garygcchiu** [comment]
For others that may run into this issue, as inspired by the repo above, their workaround/solution is to not use this plugin altogether, and specify `simdjson` as an `extraPods` in the `expo-build-properties` plugin: [[Code](https://github.com/bndkt/sharemystack/blob/4a5622c2b7265dd2f55bf0a6888d05de38525877/apps/app/app.config.ts#L76-L82)]
```
[
      "expo-build-properties",
      {
        ios: {
          extraPods: [
            {
              name: "simdjson",
              configurations: ["Debug", "Release"],
              path: "../../../node_modules/@nozbe/simdjson",
              modular_headers: true,
            },
          ],
        },
      },
    ],
```

**Comment by killerchip** [comment]
@garygcchiu Thanks for your feedback.
I'll check first chance if we can have this as an option to the plugin for easier configuration.

**Comment by atanaskanchev** [comment]
I wonder how wmdb works in [this production app ](https://github.com/bndkt/sharemystack)that is using turborepo 🤔

**Comment by killerchip** [comment]
> this production app

I think it just adds 'extra pods' for ios, without using the plugin.
On Android part, I don't think it enables JSI interface (not thoroughly checked though)

**Comment by clmoreno** [comment]
I created a PR to add monorepo support https://github.com/morrowdigital/watermelondb-expo-plugin/pull/37

**Comment by killerchip** [comment]
> I created a PR to add monorepo support #37

It is now in beta.
`yarn add @morrowdigital/watermelondb-expo-plugin@beta` to install and test it.

thanks @clmoreno

**Comment by killerchip** [comment]
This fix is now public in 2.3.2
